File: market/test/java/ru/yandex/market/adv/promo/service/environment/EnvironmentServiceTest.java
package ru.yandex.market.adv.promo.service.environment;

import java.util.Optional;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import ru.yandex.market.adv.promo.FunctionalTest;
import ru.yandex.market.common.test.db.DbUnitDataSet;

import static ru.yandex.market.adv.promo.service.environment.constant.EnvironmentSettingConstants.PROMO_MAX_COUNT;

public class EnvironmentServiceTest extends FunctionalTest {

    @Autowired
    private EnvironmentService environmentService;

    @Test
    @DbUnitDataSet(
            before = "EnvironmentServiceTest/getSettingIntegerValue_invalidCast/before.csv"
    )
    @DisplayName("Кейс, когда ожидается целочисленное значение настройки окружения, а значение - другого типа")
    void getSettingIntegerValue_invalidCast() {
        Assertions.assertEquals(Optional.empty(), environmentService.getSettingIntegerValue(PROMO_MAX_COUNT));
    }

    @Test
    @DbUnitDataSet
    @DisplayName("Кейс, когда настройка окружения ещё не была заведена")
    void get_notFound() {
        Assertions.assertEquals(Optional.empty(), environmentService.getSettingIntegerValue(PROMO_MAX_COUNT));
    }

    @Test
    @DbUnitDataSet(before = "EnvironmentServiceTest/getBooleanTrue/before.csv")
    @DisplayName("Кейс, когда из окружения берётся логическая переменная со значением true")
    void getBooleanTrue() {
        Assertions.assertEquals(Optional.of(true), environmentService.getSettingsBooleanValue("testbool"));
    }

    @Test
    @DbUnitDataSet(before = "EnvironmentServiceTest/getBooleanFalse/before.csv")
    @DisplayName("Кейс, когда из окружения берётся логическая переменная со значением false")
    void getBooleanFalse() {
        Assertions.assertEquals(Optional.of(false), environmentService.getSettingsBooleanValue("testbool"));
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/orderservice/tms/service/pg/EnvironmentServiceTest.kt
package ru.yandex.market.mbi.orderservice.tms.service.pg

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Timeout
import org.springframework.beans.factory.annotation.Autowired
import ru.yandex.market.common.test.db.DbUnitDataSet
import ru.yandex.market.mbi.orderservice.common.service.pg.EnvironmentService
import ru.yandex.market.mbi.orderservice.tms.FunctionalTest
import java.time.Duration
import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

@DbUnitDataSet(before = ["environment-service-test-before.csv"])
class EnvironmentServiceTest : FunctionalTest() {

    @Autowired
    lateinit var environmentService: EnvironmentService

    @Test
    fun testGetBoolean() {
        Assertions.assertThat(environmentService.getBoolean("boolKey1")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey2")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey3")).isTrue
        Assertions.assertThat(environmentService.getBoolean("boolKey4")).isFalse
    }

    @Test
    fun testGetLong() {
        Assertions.assertThat(environmentService.getLong("longKey1")).isEqualTo(123456L)
        Assertions.assertThatExceptionOfType(NumberFormatException::class.java).isThrownBy {
            environmentService.getLong("longKey2")
        }
    }

    @Test
    fun testSetValue() {
        Assertions.assertThat(environmentService.getString("testKey1")).isEqualTo("value1")
        environmentService.set("testKey1", "value2")
        Assertions.assertThat(environmentService.getString("testKey1")).isEqualTo("value2")
    }

    @Test
    @Timeout(1, unit = TimeUnit.SECONDS)
    fun testNotifyChanged() {
        val latch = CountDownLatch(1)
        environmentService.runOnValueChanged("trackedKey", Duration.ofMillis(100)) {
            latch.countDown()
        }
        environmentService.set("trackedKey", "abracadabra")
    }
}
